### ISSUE TITLE
[AX] Support favicon versions

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#898]
Before I start, do you guys like the new "[AX]" tag I came up with for "**A**dministrator E**x**perience"?? :smile:

See there's this annoying issue that when you upload a (new) favicon for the website, the browser tab still shows the drupal/backdrop one (or the old version of your favicon if you had one). Nothing works to show you that the favicon was indeed updated no matter how many times you refresh the page or if you clear all (website) caches.

If you browse the site with another browser that you happened not to have browsed it with before (so no favicon cache), than you get the new favicon - but only in that browser. If you ask someone to test the website and check the favicon, if they haven't browsed it before, they'll get the new version. If they did visit the site before, they'll get in the same WTF situation.

What solves the issue is a browser cache clear, but I hate that because in some browsers it also clears browsing history etc. Also this scenario:

Site owner: Oh, I saw the x website and they have their pretty logo as a cool little thingy that shows in the browser tab. Can we have that too please?
Site admin: Sure, just gimme 1min. ...[goes to create favicon from logo, logins to their site and finally uploads the favicon and saves the form]... All done!!!
Site owner: Wow, impressive!! Let me refresh the page... Nope, don't see it.(?)
Site admin: Let me check again ...[refreshes browser - realizes issue - WTF?]... ummm... gimme a few minutes to see what's wrong and I'll call you back.
Site owner: Sure. Talk to you later. [thinks something might be wrong with the site or the site admin has no skills]
Site admin: [spends x amount of time googling and reading forums for an answer, then calls back]...
 Oh, it's very simple... your browser cache needs to be cleared.
Site owner: Ummm, and how do I do that?
Site admin: I'll guide you through the steps... Go to menu x, click tab y, then button z...
Site owner: [thinking that all their site visitors will have to do that too?] Oh, I didn't know you had to do that! [actually meaning to say "Whaaaat? This guy is a looser- shouldn't have hired him."].
Site admin: Oh yes, it is a well known browser issue... [goes on to explain the technical details behind the issue]...
Site owner: Thanx for explaining all that. [actually meant "Thanx - A LOT" - not in the good sense].
Site admin: [gets the hint behind owner's voice - thinks to himself "Fucking Drupal/Bckdrop they can't get a simple thing right"]

Get the picture? :stuck_out_tongue_winking_eye: 

Anyways... solution? ...add versioning to the favicon tag and increment the version number with each new favicon uploaded:

```<link rel="shortcut icon" href="http://www.yoursite.com/favicon.ico?v=2" />```

**Comment by quicksketch** [comment]
I think this is a great idea. It's the same trick we already use to cache-bust CSS and JS files when aggregation is turned off.

> Before I start, do you guys like the new "[AX]" tag I came up with for "Administrator Experience"??

Seems like AX would be a subset of UX (as all administrators are users). Perhaps we should keep it to UX to avoid creating new Backdrop-specific vocabulary.

**Comment by klonos** [comment]
> Seems like AX would be a subset of UX (as all administrators are users). Perhaps we should keep it to UX to avoid creating new Backdrop-specific vocabulary.

Yeah, I was just being a smart ass as always - trying to get some attention and make youz smile a bit.

Changing issue title...

**Comment by quicksketch** [comment]
Thanks! :wink:

**Comment by BWPanda** [comment]
![The future is now!](https://www.tytsjerkontour.nl/images/The_Future_Is_Now_850x265.png)

Let's do this!

**Comment by klonos** [comment]
> Bumping this again. Do we need a PR?

Thanks, and yes :) ...I'd like to take a stab at this.
